test/metabase/lib/test_util.cljc

```diff
@@ -0,0 +1,64 @@
+(ns metabase.lib.test-util
+  "Misc test utils for Metabase lib."
+  (:require
+   [clojure.test :refer [is]]
+   [malli.core :as mc]
+   [medley.core :as m]
+   [metabase.lib.metadata.protocols :as metadata.protocols]
+   [metabase.lib.schema :as lib.schema]
+   [metabase.lib.test-metadata :as meta]))
+
+(def venues-query
+  {:lib/type     :mbql/query
+   :lib/metadata meta/metadata-provider
+   :type         :pipeline
+   :database     (meta/id)
+   :stages       [{:lib/type     :mbql.stage/mbql
+                   :lib/options  {:lib/uuid (str (random-uuid))}
+                   :source-table (meta/id :venues)}]})
+
+(defn venues-query-with-last-stage [m]
+  (let [query (update-in venues-query [:stages 0] merge m)]
+    (is (mc/validate ::lib.schema/query query))
+    query))
+
+(defn field-clause
+  ([table field]
+   (field-clause table field nil))
+  ([table field options]
+   [:field
+    (merge {:base-type (:base_type (meta/field-metadata table field))
+            :lib/uuid  (str (random-uuid))}
+           options)
+    (meta/id table field)]))
+
+(defn mock-metadata-provider
```

Comment:
Mock metadata provider for those rare cases when we need to test stuff not covered by `test-metadata/provider`, e.g. nested columns